src/main/java/com/mraof/minestuck/fluid/MSFluidTypes.java

```diff
@@ -0,0 +1,244 @@
+package com.mraof.minestuck.fluid;
+
+import com.mraof.minestuck.Minestuck;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.sounds.SoundEvents;
+import net.minecraft.world.entity.Entity;
+import net.minecraftforge.client.extensions.common.IClientFluidTypeExtensions;
+import net.minecraftforge.common.SoundActions;
+import net.minecraftforge.fluids.FluidType;
+import net.minecraftforge.registries.DeferredRegister;
+import net.minecraftforge.registries.ForgeRegistries;
+import net.minecraftforge.registries.RegistryObject;
+
+import java.util.function.Consumer;
+
+public class MSFluidTypes
+{
+	public static final DeferredRegister<FluidType> REGISTER = DeferredRegister.create(ForgeRegistries.Keys.FLUID_TYPES, Minestuck.MOD_ID);
+	
+	//TODO fix flowing fluid push mechanics (player is too slow to walk up stream)
```

Comment:
Not sure where would be best to change this to accommodate the problem

Reply:
Still unsure how to fix this without also resulting in changes to movement in source blocks